lib/dal/src/migrations/U2612__avoid_redundant_work_during_component_creation.sql

```diff
@@ -0,0 +1,382 @@
+            ident_nil_v1(),
+            ident_nil_v1(),
+            this_new_row.id
+        );
+
+        SELECT ap.*
+        INTO STRICT this_attribute_prototype
+        FROM attribute_prototypes_v1(this_tenancy, this_visibility) AS ap
+            INNER JOIN attribute_value_belongs_to_attribute_prototype_v1(this_tenancy, this_visibility) AS avbtap
+                ON ap.id = avbtap.belongs_to_id
+        WHERE avbtap.object_id = this_prop_attribute_value.id;
+
+        SELECT av.object
+        INTO this_new_attribute_value
+        FROM attribute_value_create_v1(
+            this_tenancy,
+            this_visibility,
+            this_attribute_context,
+            this_prop_attribute_value.func_binding_id,
+            this_prop_attribute_value.func_binding_return_value_id,
+            this_prop_attribute_value.key
+        ) AS av;
+
+        PERFORM set_belongs_to_v1(
+            'attribute_value_belongs_to_attribute_prototype',
+            this_tenancy,
+            this_visibility,
+            this_new_attribute_value ->> 'id',
+            this_attribute_prototype.id
+        );
+    END LOOP;
+
+    -- We need to create the attribute_value_belongs_to_attribute_value
+    -- relationship for the Prop Attribute Values of the Component. We are doing
+    -- this after all of the Attribute Values have been created because we're
+    -- guaranteeing that they're created in topographical order, which prevents
+    -- us from setting the belongs_to relationship as we go along.
+    this_attribute_context := attribute_context_build_from_parts_v1(
+        NULL, -- Prop ID
+        ident_nil_v1(), -- Internal Provider ID
+        ident_nil_v1(), -- External Provider ID
+        this_new_row.id -- Component ID
+    );
+    FOR this_parent_attribute_value_id, this_attribute_value_id IN
+        WITH RECURSIVE avbtav(parent_av_id, av_id) AS (
+            SELECT parent_av.id, av.id
+            FROM prop_belongs_to_prop AS pbtp
+            INNER JOIN attribute_values_v1(this_tenancy, this_visibility) AS av
+                ON av.attribute_context_prop_id = pbtp.object_id
+            INNER JOIN attribute_values_v1(this_tenancy, this_visibility) AS parent_av
+                ON parent_av.attribute_context_prop_id = pbtp.belongs_to_id
+            WHERE in_attribute_context_v1(this_attribute_context, av)
+                AND av.attribute_context_component_id = this_new_row.id
+                AND in_attribute_context_v1(this_attribute_context, parent_av)
+                AND parent_av.attribute_context_component_id = this_new_row.id
+        )
+        SELECT * FROM avbtav
+    LOOP
+        PERFORM set_belongs_to_v1(
+            'attribute_value_belongs_to_attribute_value',
+            this_tenancy,
+            this_visibility,
+            this_attribute_value_id,
+            this_parent_attribute_value_id
+        );
+    END LOOP;
+
+    -- -- Make sure we've populated the dependency graph for this (new) component.
```

Comment:
This is the biggest change, we remove that block because this does exactly what `attribute_value_dependencies_update_component_v1` does, but without deduplicating.

So now we schedule that work during create so it can be deduplicated. I'm observing Component::new go from 7s locally to 1s, those 6s are exactly the same time `attribute_value_dependencies_update_component_v1` takes. So we can be sure that we were duplicating that processing and now arent.